fil_actor_interface/src/builtin/multisig/mod.rs

```diff
@@ -84,7 +99,71 @@ impl State {
             State::V11(st) => from_token_v3_to_v2(st.amount_locked(height)),
-            State::V12(st) => from_token_v3_to_v2(st.amount_locked(height)),
+            State::V12(st) => from_token_v4_to_v2(st.amount_locked(height)),
         })
     }
+
+    /// Returns pending transactions for the given multisig wallet
+    pub fn get_pending_txn<BS: Blockstore>(&self, store: &BS) -> anyhow::Result<Vec<Transaction>> {
+        let mut res = Vec::new();
+        match self {
+            State::V8(st) => {
+                let txns = fil_actors_shared::v8::make_map_with_root::<
+                    BS,
+                    fil_actor_multisig_state::v8::Transaction,
+                >(&st.pending_txs, store)?;
+                parse_pending_transactions!(res, txns);
+                Ok(res)
+            }
+            State::V9(st) => {
+                let txns = fil_actors_shared::v9::make_map_with_root::<
+                    BS,
+                    fil_actor_multisig_state::v9::Transaction,
+                >(&st.pending_txs, store)?;
+                parse_pending_transactions!(res, txns);
+                Ok(res)
+            }
+            State::V10(st) => {
+                let txns = fil_actors_shared::v10::make_map_with_root::<
+                    BS,
+                    fil_actor_multisig_state::v10::Transaction,
+                >(&st.pending_txs, store)?;
+                parse_pending_transactions!(res, txns, from_address_v3_to_v2, from_token_v3_to_v2);
+                Ok(res)
+            }
+            State::V11(st) => {
+                let txns = fil_actors_shared::v11::make_map_with_root::<
+                    BS,
+                    fil_actor_multisig_state::v11::Transaction,
+                >(&st.pending_txs, store)?;
+                parse_pending_transactions!(res, txns, from_address_v3_to_v2, from_token_v3_to_v2);
+                Ok(res)
+            }
+            State::V12(st) => {
+                let txns = fil_actor_multisig_state::v12::PendingTxnMap::load(
+                    store,
+                    &st.pending_txs,
+                    fil_actor_multisig_state::v12::PENDING_TXN_CONFIG,
+                    "pending txns",
+                )
+                .expect("Could not load pending transactions");
+                txns.for_each(|tx_id, txn| {
+                    res.push(Transaction {
+                        id: tx_id.0,
+                        to: from_address_v4_to_v2(txn.to),
+                        value: from_token_v4_to_v2(txn.value.clone()),
+                        method: txn.method,
+                        params: txn.params.clone(),
+                        approved: txn
+                            .approved
+                            .iter()
+                            .map(|&addr| from_address_v4_to_v2(addr))
+                            .collect(),
+                    });
+                    Ok(())
```

Comment:
Could this be hidden behind a macro/function to keep it consistent with the variants above?

Reply:
done